chunk_downloader.go

```diff
@@ -630,3 +681,32 @@ func copyChunkStream(body io.Reader, rows chan<- []*string) error {
 	}
 	return nil
 }
+
+// ResultBatch object represents a chunk of data, or subset of rows, retrievable in array.Record format
+type ResultBatch struct {
+	Rec                *[]array.Record
+	idx                int
+	scd                *snowflakeChunkDownloader
+	funcDownloadHelper func(context.Context, *snowflakeChunkDownloader, int) error
+}
```

Comment:
Can we expose some metadata such as rowcount and size information that we retrieve from the server?

Reply:
adding metadata for rowcount. For size it does not seem to be achievable given that we read the arrow.Records right away and those do not contain any metadata on size themselves.